src/CurveShutdownStrategy.sol

```diff
@@ -0,0 +1,136 @@
+▒ ▓██░ ▒░▒██▀▀██░▒███      ▒██▀▀██░▒███   ▓██ ░▄█ ▒░██   █▌
+░ ▓██▓ ░ ░▓█ ░██ ▒▓█  ▄    ░▓█ ░██ ▒▓█  ▄ ▒██▀▀█▄  ░▓█▄   ▌
+  ▒██▒ ░ ░▓█▒░██▓░▒████▒   ░▓█▒░██▓░▒████▒░██▓ ▒██▒░▒████▓ 
+  ▒ ░░    ▒ ░░▒░▒░░ ▒░ ░    ▒ ░░▒░▒░░ ▒░ ░░ ▒▓ ░▒▓░ ▒▒▓  ▒ 
+    ░     ▒ ░▒░ ░ ░ ░  ░    ▒ ░▒░ ░ ░ ░  ░  ░▒ ░ ▒░ ░ ▒  ▒ 
+  ░       ░  ░░ ░   ░       ░  ░░ ░   ░     ░░   ░  ░ ░  ░ 
+          ░  ░  ░   ░  ░    ░  ░  ░   ░  ░   ░        ░    
+                                                    ░      
+              .,;>>%%%%%>>;,.
+           .>%%%%%%%%%%%%%%%%%%%%>,.
+         .>%%%%%%%%%%%%%%%%%%>>,%%%%%%;,.
+       .>>>>%%%%%%%%%%%%%>>,%%%%%%%%%%%%,>>%%,.
+     .>>%>>>>%%%%%%%%%>>,%%%%%%%%%%%%%%%%%,>>%%%%%,.
+   .>>%%%%%>>%%%%>>,%%>>%%%%%%%%%%%%%%%%%%%%,>>%%%%%%%,
+  .>>%%%%%%%%%%>>,%%%%%%>>%%%%%%%%%%%%%%%%%%,>>%%%%%%%%%%.
+  .>>%%%%%%%%%%>>,>>>>%%%%%%%%%%'..`%%%%%%%%,;>>%%%%%%%%%>%%.
+.>>%%%>>>%%%%%>,%%%%%%%%%%%%%%.%%%,`%%%%%%,;>>%%%%%%%%>>>%%%%.
+>>%%>%>>>%>%%%>,%%%%%>>%%%%%%%%%%%%%`%%%%%%,>%%%%%%%>>>>%%%%%%%.
+>>%>>>%%>>>%%%%>,%>>>%%%%%%%%%%%%%%%%`%%%%%%%%%%%%%%%%%%%%%%%%%%.
+>>%%%%%%%%%%%%%%,>%%%%%%%%%%%%%%%%%%%'%%%,>>%%%%%%%%%%%%%%%%%%%%%.
+>>%%%%%%%%%%%%%%%,>%%%>>>%%%%%%%%%%%%%%%,>>%%%%%%%%>>>>%%%%%%%%%%%.
+>>%%%%%%%%;%;%;%%;,%>>>>%%%%%%%%%%%%%%%,>>>%%%%%%>>;";>>%%%%%%%%%%%%.
+`>%%%%%%%%%;%;;;%;%,>%%%%%%%%%>>%%%%%%%%,>>>%%%%%%%%%%%%%%%%%%%%%%%%%%.
+ >>%%%%%%%%%,;;;;;%%>,%%%%%%%%>>>>%%%%%%%%,>>%%%%%%%%%%%%%%%%%%%%%%%%%%%.
+ `>>%%%%%%%%%,%;;;;%%%>,%%%%%%%%>>>>%%%%%%%%,>%%%%%%'%%%%%%%%%%%%%%%%%%%>>.
+  `>>%%%%%%%%%%>,;;%%%%%>>,%%%%%%%%>>%%%%%%';;;>%%%%%,`%%%%%%%%%%%%%%%>>%%>.
+   >>>%%%%%%%%%%>> %%%%%%%%>>,%%%%>>>%%%%%';;;;;;>>,%%%,`%     `;>%%%%%%>>%%
+   `>>%%%%%%%%%%>> %%%%%%%%%>>>>>>>>;;;;'.;;;;;>>%%'  `%%'          ;>%%%%%>
+    >>%%%%%%%%%>>; %%%%%%%%>>;;;;;;''    ;;;;;>>%%%                   ;>%%%%
+    `>>%%%%%%%>>>, %%%%%%%%%>>;;'        ;;;;>>%%%'                    ;>%%%
+     >>%%%%%%>>>':.%%%%%%%%%%>>;        .;;;>>%%%%                    ;>%%%'
+     `>>%%%%%>>> ::`%%%%%%%%%%>>;.      ;;;>>%%%%'                   ;>%%%'
+      `>>%%%%>>> `:::`%%%%%%%%%%>;.     ;;>>%%%%%                   ;>%%'
+       `>>%%%%>>, `::::`%%%%%%%%%%>,   .;>>%%%%%'                   ;>%'
+        `>>%%%%>>, `:::::`%%%%%%%%%>>. ;;>%%%%%%                    ;>%,
+         `>>%%%%>>, :::::::`>>>%%%%>>> ;;>%%%%%'                     ;>%,
+          `>>%%%%>>,::::::,>>>>>>>>>>' ;;>%%%%%                       ;%%,
+            >>%%%%>>,:::,%%>>>>>>>>'   ;>%%%%%.                        ;%%
+             >>%%%%>>``%%%%%>>>>>'     `>%%%%%%.
+             >>%%%%>> `@@a%%%%%%'     .%%%%%%%%%.
+             `a@@a%@'    `%a@@'       `a@@a%a@@a
+ */
+
+import "src/CRVStrategy.sol";
+
+/// @notice Strategy contract, supporting Shutdown.
+contract CurveShutdownStrategy is CRVStrategy {
+    /// @notice Mapping of shutdown gauges.
+    mapping(address => bool) public isShutdown;
+
+    /// @notice Mapping of protected gauges.
+    mapping(address => bool) public protectedGauges;
```

Comment:
I would recommend merging both enums into a single `mapping` and using an enum to track the status of a gauge. This approach not only improves gas efficiency but also helps prevent future bugs that could result in the gauge being stored in both mappings simultaneously. By utilizing one mapping and an enum, you create a simple state machine that significantly reduces the likelihood of such bugs, regardless of how the code evolves in the future. The state of each gauge is also easier to track, whether you need it on-chain or off-chain.

```solidity
enum Status {
		Default,	// 0 -- Default
		Shutdown,	// 1
		Protected,	// 2
}

mapping(address => Status) public gaugesStatus;
```

Setters and getters are also straightforward to implement using enums.

Reply:
Same comment for `YearnShutdownStrategy`.